agent/run_job.go

```diff
@@ -428,6 +433,50 @@ func (r *JobRunner) cleanup(ctx context.Context, wg *sync.WaitGroup, exit core.P
 	r.agentLogger.Info("Finished job %s for build at %s", r.conf.Job.ID, r.conf.Job.Env["BUILDKITE_BUILD_URL"])
 }
 
+// waitForKubernetesProcessesToComplete waits for Kubernetes command containers to finish
+// before stopping log collection, ensuring post-command hook logs are captured.
+func (r *JobRunner) waitForKubernetesProcessesToComplete(ctx context.Context) {
+	r.agentLogger.Debug("[JobRunner] Waiting for Kubernetes processes to complete before stopping log collection")
+
+	// Guard against nil process
+	if r.process == nil {
+		r.agentLogger.Debug("[JobRunner] No process to wait for")
+		return
+	}
+
+	// Wait for the process to actually start before waiting for it to complete
+	select {
+	case <-r.process.Started():
+		// Process has started, we can now safely wait for completion
+	case <-ctx.Done():
+		r.agentLogger.Debug("[JobRunner] Context cancelled before process started, skipping wait")
+		return
+	}
+
+	gracePeriod := r.conf.AgentConfiguration.KubernetesLogCollectionGracePeriod
+
+	waitCtx := ctx
+	if gracePeriod > 0 {
```

Comment:
What if, for some reason, someone wanted to disable the grace period? I think `0` should probably mean `0`. What about:


```suggestion
	if gracePeriod >= 0 {
```